lib/solvers/AutoroutingPipelineSolver.ts

```diff
@@ -122,8 +126,80 @@ export class AutoroutingPipelineSolver extends BaseSolver {
   srjWithPointPairs?: SimpleRouteJson
   capacityNodes: CapacityMeshNode[] | null = null
   capacityEdges: CapacityMeshEdge[] | null = null
+  capacityEdgesWithPortals: CapacityMeshEdge[] | null = null
 
   cacheProvider: CacheProvider | null = null
+  /**
+   * Removes synthetic offboard edges from the capacity edges list.
+   */
+  private stripOffboardEdges(stage: string) {
+    if (!this.capacityEdges) return
+    const originalCount = this.capacityEdges.length
+    this.capacityEdges = this.capacityEdges.filter(
+      (edge) => !edge.capacityMeshEdgeId.startsWith("offboard-"),
+    )
+    const removed = originalCount - this.capacityEdges.length
+    this.capacityEdgesWithPortals = null
+  }
```

Comment:
don't pollute top level functions/classes. This function is too specific to be in this class- this pipeline is vary important so we should try to keep utility functions out of this file